common/src/main/java/com/graphicsfuzz/common/util/SideEffectChecker.java

```diff
@@ -43,24 +43,26 @@ private static boolean isSideEffectFreeVisitor(IAstNode node,
 
       @Override
       public void visitFunctionCallExpr(FunctionCallExpr functionCallExpr) {
-        if (!TyperHelper.getBuiltins(shadingLanguageVersion, shaderKind)
+        if (TyperHelper.getBuiltins(shadingLanguageVersion, shaderKind)
```

Comment:
So this is where the bug was?

Reply:
No, the bug was in ReductionOpportunitiesBase, and was to do with when entering and leaving dead code injections was being tracked.

The fix led to some test failures, some of which meant changing the tests (as explained in the PR description). But a few could be made to pass again by having this refinement to the side effect checker.